### PR TITLE
Add wipertools

### PR DESCRIPTION
**Pull request by mazzalab**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/wipertools/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - fastqwiper
+  commands:
+    - wipertools --help
+
+about:
+  home: https://github.com/mazzalab/fastqwiper
+  summary: A suite of programs that drop or fix pesky lines in FASTQ files and that split FASTQ files into chunks or merge them.
+  license: CC-BY-NC-ND-4.0
+  license_family: CC
+  license_file: LICENSE.txt
```
[the remark this answers is not in the document]
Comment (irongraft) [reply recipes/wipertools/meta.yaml:41]: I've changed the license to GPL. It should be fine now

**Comment by mazzalab** [comment]
@BiocondaBot please fetch artifacts

**Comment by mazzalab** [comment]
@BiocondaBot please add label

**Review by irongraft — APPROVED** [review]
I've checked locally and it seems to work

File: recipes/wipertools/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+source:
+  url: https://github.com/mazzalab/fastqwiper/archive/refs/tags/v{{ version }}.zip
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  noarch: python
+  run_exports:
+    - {{ pin_subpackage('wipertools', max_pin="x.x") }}
+  entry_points:
```
[the remark this answers is not in the document]
Comment (irongraft) [reply recipes/wipertools/meta.yaml]: You gave a wrong suggestion: removing run_exports led to: ERROR: recipes/wipertools/meta.yaml:0: missing_run_exports: Recipe should have a run_exports statement that ensures correct pinning in downstream packages
Error: Process completed with exit code 1.

File: recipes/wipertools/build.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+# mkdir -p $PREFIX/lib/python$PY_VER/site-packages/tests/testdata
+# cp -r tests/testdata/* $PREFIX/lib/python$PY_VER/site-packages/tests/testdata/
+mkdir -p $PREFIX/lib/python$PY_VER/site-packages/
+cp versions.json $PREFIX/lib/python$PY_VER/site-packages/
```
[the remark this answers is not in the document]
Comment (mazzalab) [reply recipes/wipertools/build.sh:6]: versions.json is present in https://github.com/mazzalab/fastqwiper/archive/refs/tags/v1.1.0.zip. I do not know why you do not find it inside the zip file.

[the remark this answers is not in the document]
Comment (mazzalab) [reply recipes/wipertools/build.sh:6]: OK. I've added MANIFEST.IN and include_package_data=True

**Comment by mazzalab** [comment]
@BiocondaBot please fetch artifacts

**Comment by mazzalab** [comment]
@bioconda/core can I have a review for this PR?

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks @mazzalab

**Pull request merged**